components/dashboard/ExpenseReporter/index.tsx

```diff
@@ -0,0 +1,46 @@
+import { gql, useQuery } from '@apollo/client'
+import React from 'react'
+import { Bar, BarChart, CartesianGrid, Legend, ResponsiveContainer, Tooltip, XAxis, YAxis } from 'recharts'
+
+import DateHelper from '../../../lib/date-helper'
+
+function ExpenseReporter() {
+  const today = new Date()
+  const lastSixMonth = DateHelper.dateBeforeSixMonths()
+
+  const urlLink = `?startmonth=${today.getFullYear()}${today.getMonth()}&endmonth=${lastSixMonth.getFullYear()}${lastSixMonth.getMonth()}`
+
+  const mutateQuery = (urlParams: string) => gql`
+    query MajorPlannedExpenses {
```

Comment:
move this out of the component

Reply:
Couldn't find much documentation on how to call apollo client in nextjs, but here: https://www.apollographql.com/blog/apollo-client/next-js/next-js-getting-started/
It's recommended to make graphql query calls in individual components.

might later remove it once a better functional way is available